### PR TITLE
Update the encrypted backup details on encryption algorithms

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-8486

A customer recently asked for more detail about the encryption algorithms we use for encrypted backups. This PR adds details on this to the section already present.

### CODE REVIEW

File: src/current/v23.1/take-and-restore-encrypted-backups.md [hunk 1]
```diff
@@ -20,13 +20,15 @@ This doc provides information about how to take and restore encrypted backups in
...
+The backup-specific random key is then encrypted using the Key Management Service (KMS) APIs provided by the relevant cloud provider—AWS KMS, Google Cloud KMS, or Azure Key Vault. The resulting encrypted version of the random key is then stored within the backup metadata. Note that the encryption algorithm for the random key is determined by the specific cloud provider, with [AWS](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#symmetric-cmks), [Azure](https://learn.microsoft.com/en-us/azure/security/fundamentals/encryption-atrest), and [GCP](https://cloud.google.com/kms/docs/algorithms#symmetric_encryption_algorithms) using AES-GCM-256 for symmetric encryption.
```
Comment (kathancox) [inline src/current/v23.1/take-and-restore-encrypted-backups.md:27]: I assume that Azure also uses AES-GCM-256?

**Comment by kathancox** [comment]
@stevendanna Sorry to tag you on another review, this came up as a customer issue a couple of months ago, and I am using the summary you gave to support for the update (hopefully correctly!).

File: src/current/v23.1/take-and-restore-encrypted-backups.md [hunk 2]
```diff
@@ -20,13 +20,15 @@ This doc provides information about how to take and restore encrypted backups in
...
 
-On [`RESTORE`]({% link {{ page.version.version }}/restore.md %}), CockroachDB reads the `ENCRYPTION_INFO` file and attempts to decrypt the encrypted data key using the KMS URI provided in the `RESTORE` statement. Once CockroachDB successfully obtains the unencrypted data key, the `BACKUP` manifest and data files will be decrypted and the restoration will proceed. Similarly, the same KMS URI is needed to decrypt the file to list the contents of the backup when using [`SHOW BACKUP`]({% link {{ page.version.version }}/show-backup.md %}#show-an-encrypted-backup).
+During the [backup process]({% link {{ page.version.version }}/backup-architecture.md %}), a cryptographically secure 32-byte (256-bit) key is generated. This backup-specific random key is then encrypted using the Key Management Service (KMS) APIs provided by the relevant cloud provider—AWS KMS, Google Cloud KMS, or Azure Key Vault. The resulting encrypted version of the random key is then stored within the backup metadata in the `ENCRYPTION_INFO` file.
```
Comment (kathancox) [inline src/current/v23.1/take-and-restore-encrypted-backups.md:25]: @stevendanna what do you think? Is the `ENCRYPTION_INFO` file bit here correct? I felt like it was important to still include that, as practically, that is what the user will see in their storage bucket.

**Review by stevendanna — APPROVED** [review]

File: src/current/v23.1/take-and-restore-encrypted-backups.md [hunk 3]
```diff
@@ -20,13 +20,15 @@ This doc provides information about how to take and restore encrypted backups in
...
+During a restore job, CockroachDB retrieves the encrypted random key from the backup metadata and attempts to decrypt it using the KMS URI specified in the [`RESTORE`]({% link {{ page.version.version }}/restore.md %}) statement. Once successfully decrypted, CockroachDB uses this key to decrypt the [`BACKUP`]({% link {{ page.version.version }}/backup.md %}) manifest and data files. Similarly, the same KMS URI is required for decrypting the files when listing the backup contents using [`SHOW BACKUP`]({% link {{ page.version.version }}/show-backup.md %}).
+
+When incremental backups are in use, the `KMS` option is applied to all backup file URLs. Therefore, each incremental backup must include at least one of the KMS URIs used during the full backup. This subset can consist of any combination of the original URIs, but you cannot introduce new KMS URIs. Likewise, when taking [locality-aware backups](#locality-aware-backup-with-kms-encryption), the specified KMS URI is applied to files across all localities.
```
Comment (florence-crl) [inline src/current/v23.1/take-and-restore-encrypted-backups.md:31]: Should `KMS` be lower case `kms`? Or can it be either case? The examples have both, but the [syntax page](https://deploy-preview-18211--cockroachdb-docs.netlify.app/docs/v23.1/backup#options) has lower case.

**Review by florence-crl — APPROVED** [review]
LGTM with 1 consistency nit

**Pull request merged**